### ISSUE TITLE
Customer getting charged twice when error occurs on first try

### ISSUE THREAD

**Issue by runna-app** [RevenueCat/react-native-purchases#319]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/react-native-purchases/blob/main/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/react-native-purchases/issues)

**Describe the bug**
We have had 3 instances where a user has been charged twice for a `Consumable Product`. We have noticed that this happens when the user receives an error (example of error response below) from the API method `await Purchases.purchaseProduct(productId, null, Purchases.PURCHASE_TYPE.INAPP)`. Below is a snippet of how we use the API method in a try/catch block. As you can see we are displaying an error model when the error code is not equal to `1`. This is because the error code will be equal to `1` when a user presses `cancel` on the native IAP model. 

```
                         try {
				setIapLoading(true)
				await Purchases.setAttributes(userAttributes)
				log.info('Set user attributed', userAttributes)
				await Purchases.purchaseProduct(productId, null, Purchases.PURCHASE_TYPE.INAPP)
				log.info('Purchase successful')
				navigation.navigate(ROUTE_SIGN_UP)
			} catch (e: any) {
				if (e.code !== '1') {
					log.error('IAP Error for Product: ' + productId, JSON.stringify(e))
					setIsOpen(true)
				}
			}
```

Error Response:
```
{"code":"2","message":"There was a problem with the App Store.","domain":"RCPurchasesErrorDomain","userInfo":{"readable_error_code":"STORE_PROBLEM","readableErrorCode":"STORE_PROBLEM","NSLo...
```

When the above error occurs we catch the error in a catch block and allow the user to retry purchasing. However, this is causing the product to be purchased twice. Below is a screenshot of the customer history in the RC dashboard.

![image](https://user-images.githubusercontent.com/77796515/147455610-70662e9b-82de-4c05-abdb-032fd40af4d3.png)

1. Environment
   1. Platform: `iOS`
   2. SDK version: `^4.5.0`
   3. OS version: `iOS 15.1.1`
   4. Xcode/Android Studio version: `xcode 13.1`
   5. React Native version: `0.66.4`
   6. SDK installation (CocoaPods + version or manual): `CocoaPods`
   7. How widespread is the issue. Percentage of devices affected. `3 instances so far`
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
3. Steps to reproduce, with a description of expected vs. actual behavior
4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

**Additional context**

**Comment by beylmk** [comment]
Hey there, I'm going through old issues and wanted to confirm whether this one was solved? I see that you also had a support ticket around the same issue.

Let me know if it's okay to go ahead and close this.

**Comment by walterholohan** [comment]
@beylmk you can close this one. The above is expected behaviour from Apple when there is an interrupted purchase and RevenueCat has docs on how to handle this

**Comment by beylmk** [comment]
Thanks for following up!

**Comment by mklb** [comment]
I see the same happening to a small percentage of app users since I updated to V7 from V5. I have no idea why.

Today I observed an android user:

Session 1: App download -> purchases PRO.
Session 2: App start -> somehow the app purchases the same purchase twice more in the background which is undoable once you unlocked PRO. The user confirms that he did just buy it once. He reached out because he saw multiple bank charges.
